personalrobotics/ada_feeding
# Add pre-commit hooks & BSD 3-Clause license

# Description

This PR adds pre-commit hooks to this repository. This ensures consistent formatting and style throughout the repository. After the PR gets merged in, I'll be able to configure this repository to automatically run the pre-commit hooks on every PR.

Related PRs:
- [`ada_ros2`#51](https://github.com/personalrobotics/ada_ros2/pull/51)
-  [`feeding_web_interface`#150](https://github.com/personalrobotics/feeding_web_interface/pull/150)

# Testing procedure

Although in theory this should not change any functionality, we must check. Thus:

- [x] Start the demo in `real`: `python3 src/ada_Feeding/start.py`
- [x] Eat a whole bite, verify everything works as expected.
- [x] Start the demo in `mock`: `python3 src/ada_Feeding/start.py --sim mock`
- [x] Go through a whole bite, verify everything works as expected.

# Before opening a pull request
- \[x\] `pre-commit run --all-files`
- \[x\] Run your code through [pylint](https://pylint.readthedocs.io/en/latest/). `pylint --recursive=y --rcfile=.pylintrc .`. All warnings but `fixme` must be addressed.

# Before Merging
